commands/version/index.js

```diff
@@ -278,10 +278,9 @@ class VersionCommand extends Command {
     const { bump, conventionalCommits, preid } = this.options;
     const repoVersion = bump ? semver.clean(bump) : "";
     const increment = bump && !semver.valid(bump) ? bump : "";
-    const isPrerelease = increment.startsWith("pre");
 
     const getExistingPreId = version => (semver.prerelease(version) || []).shift();
-    const resolvePrereleaseId = existingPreid => preid || (isPrerelease && existingPreid) || "alpha";
```

Comment:
It is certainly distressing that removing this inline conditional doesn't break _any_ tests. The prompting _is_ very poorly tested, I'll be the first to admit.

I would expect a modification to `makePromptVersion`'s arguments to be necessary to establish the correct `preid` in the dialog, at the very least. I think the best way to write the tests is to do them first, before changing this code, just so we can establish what _fails_ in the current context. Without that, I really can't say if this is the right approach.

Reply:
Sounds good. I'll revert locally and poke around with the tests to try to find the use case that relies on the `isPrerelease` conditional.

Reply:
I've been looking at this a bit and I think that the `isPrerelease` condition is a no-op.

The result of `resolvePrereleaseId` function is the identifier passed as the third argument to `semver.inc`. The identifier is only used when the second argument is a `pre___` type; for `major`, `minor`, and `patch` releases, the identifier is ignored.

If the condition is removed, a failing test would happen when `isPrerelease` is false and `existingPreid` is not `undefined`; the third argument to `semver.inc` would be `existingPreid` instead of `alpha`. However, since the value returned by `resolvePrereleaseId` is never used if `isPrerelease` is false, the value doesn't actually matter.

Reply:
You're right, of course. Thanks for taking the time to explain it in detail. I verified it locally with this change:
```diff
--- a/commands/version/__tests__/version-bump-prerelease.test.js
+++ b/commands/version/__tests__/version-bump-prerelease.test.js
@@ -152,8 +152,8 @@ test("independent version prerelease does not bump on every unrelated change", a
   // simulate choices for pkg-a then pkg-b
   prompt.mockChoices("patch", "PRERELEASE");
   prompt.input.mockImplementationOnce((msg, cfg) =>
-    // a convoluted way of entering "bumps" at the prompt
-    Promise.resolve(cfg.filter("bumps"))
+    // the _existing_ "bumps" prerelease ID should be preserved
+    Promise.resolve(cfg.filter())
   );
 
   await lernaVersion(cwd)();
```